src/Core/Services/TypeHelper.cs

```diff
@@ -96,26 +91,117 @@ public static class TypeHelper
-        public static string SystemTypeToJsonDataType(Type type)
+        public static JsonDataType GetJsonDataTypeFromSystemType(Type type)
         {
+            // Get the underlying type argument if the 'type' argument is a nullable type.
+            Type? nullableUnderlyingType = Nullable.GetUnderlyingType(type);
+            if (nullableUnderlyingType is not null)
+            {
+                type = nullableUnderlyingType;
+            }
+
             if (!_systemTypeToJsonDataTypeMap.TryGetValue(type, out JsonDataType openApiJsonTypeName))
             {
                 openApiJsonTypeName = JsonDataType.Undefined;
             }
 
-            string formattedOpenApiTypeName = openApiJsonTypeName.ToString().ToLower();
-            return formattedOpenApiTypeName;
+            return openApiJsonTypeName;
+        }
+
+        /// <summary>
+        /// Returns the DbType for given system type.
+        /// </summary>
+        /// <param name="systemType">The system type for which the DbType is to be determined.</param>
+        /// <returns>DbType for the given system type.</returns>
+        public static DbType? GetDbTypeFromSystemType(Type systemType)
+        {
+            if (!_systemTypeToDbTypeMap.TryGetValue(systemType, out DbType dbType))
+            {
+                return null;
+            }
+
+            return dbType;
+        }
+
+        /// <summary>
+        /// Converts the string representation of a SQL Server data type to the corrsponding .NET Framework/CLR type as documented
+        /// by the SQL Server data type mappings article.
+        /// The SQL Server database engine type and SqlDbType enum map 1:1 when character casing is ignored.
+        /// e.g. SQL DB type 'bigint' maps to SqlDbType enum 'BigInt' in a case-insensitive match.
+        /// There are some mappings in the SQL Server data type mappings table which do not map after ignoring casing, however
+        /// those mappings are outdated and don't accommodate newly added SqlDbType enum values added.
+        /// e.g. The documentation table shows SQL server type 'binary' maps to SqlDbType enum 'VarBinary',
+        /// however SqlDbType.Binary now exists.
+        /// </summary>
+        /// <param name="dbTypeName">String value sourced from the DATA_TYPE column in the Procedure Parameters or Columns
+        /// schema collections.</param>
+        /// <seealso cref="https://learn.microsoft.com/dotnet/framework/data/adonet/sql-server-schema-collections#columns"/>
+        /// <seealso cref="https://learn.microsoft.com/dotnet/framework/data/adonet/sql-server-schema-collections#procedure-parameters"/>
+        /// <exception>Failed type conversion.</exception>"
+        public static Type GetSystemTypeFromSqlDbType(string sqlDbTypeName)
+        {
+            if (Enum.TryParse(enumType: typeof(SqlDbType), value: sqlDbTypeName, ignoreCase: true, out object? result) && result is not null)
```

Comment:
You can save the cast on line 193 with `Enum.TryParse<SqlDbType>`.

Reply:
You don't need to do the `null` test on `result` as `Enum.TryParse` has `NotNullWhen(true)` on the `out` parameter.